src/App.jsx

```diff
@@ -1,35 +1,65 @@
-import { useState } from 'react'
-import reactLogo from './assets/react.svg'
-import viteLogo from '/vite.svg'
-import './App.css'
+import { useState, useEffect } from "react";
+import axios from "axios";
+import CharacterList from "./components/CharacterList";
+import "./App.module.css";
 
 function App() {
-  const [count, setCount] = useState(0)
+  const [characters, setCharacters] = useState([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(null);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [hasMore, setHasMore] = useState(true);
+
+  const fetchCharacters = async (page = 1) => {
+    setIsLoading(true);
+    try {
+      const response = await axios.get(
+        `https://rickandmortyapi.com/api/character?page=${page}`
+      );
```

Comment:
1. API URL must be in .env file, read about enviroment variables

2. Configure axios instance, to make it reusable.